contract/src/main/java/io/neow3j/contract/NefFile.java

```diff
@@ -0,0 +1,200 @@
+package io.neow3j.contract;
+
+import io.neow3j.constants.NeoConstants;
+import io.neow3j.crypto.Hash;
+import io.neow3j.io.BinaryReader;
+import io.neow3j.io.BinaryWriter;
+import io.neow3j.io.IOUtils;
+import io.neow3j.io.NeoSerializable;
+import io.neow3j.io.exceptions.DeserializationException;
+import io.neow3j.utils.ArrayUtils;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * +------------+-----------+------------------------------------------------------------+
+ * |   Field    |  Length   |                          Comment                           |
+ * +------------+-----------+------------------------------------------------------------+
+ * | Magic      | 4 bytes   | Magic header                                               |
+ * | Compiler   | 32 bytes  | Compiler used                                              |
+ * | Version    | 16 bytes  | Compiler version (Mayor, Minor, Build, Version)            |
+ * | ScriptHash | 20 bytes  | ScriptHash for the script                                  |
+ * +------------+-----------+------------------------------------------------------------+
+ * | Checksum   | 4 bytes   | Sha256 of the header (CRC)                                 |
+ * +------------+-----------+------------------------------------------------------------+
+ * | Script     | Var bytes | Var bytes for the payload                                  |
+ * +------------+-----------+------------------------------------------------------------+
+ */
+public class NefFile extends NeoSerializable {
+
+    // NEO Executable Format 3 (NEF3)
+    private static final int MAGIC = 0x3346454E;
```

Comment:
haha, it's the "magic". 😄

Reply:
Neo core uses a magic number here and also when creating hashes e.g. for signature generation. For lack of a better name i went with the same.